dvc/stage/imports.py

```diff
@@ -39,5 +56,13 @@ def sync_import(stage, dry=False, force=False, jobs=None):
     if not force and stage.already_cached():
         stage.outs[0].checkout()
     else:
+        if check_changed:
+            old_hash_info = stage.deps[0].hash_info
         stage.save_deps()
-        stage.deps[0].download(stage.outs[0], jobs=jobs)
+        if check_changed and not old_hash_info == stage.deps[0].hash_info:
```

Comment:
For the record (unrelated to this PR, it does it correctly for now): in the short future this should really be a meta check rather than hash_info, since `etag`s are really not hashes.